### PR TITLE
Update diskmaker-x for catalina

### PR DESCRIPTION
**Pull request by patbaumgartner**
Adding as discussed a fallback strategy for mojave.

After making all changes to the cask:

- [ x] `brew cask audit --download {{cask_file}}` is error-free.
- [ x] `brew cask style --fix {{cask_file}}` reports no offenses.
- [ x] The commit message includes the cask’s name and version.
- [ x] The submission is for [a stable version](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask) or [documented exception](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#but-there-is-no-stable-version).

### CODE REVIEW

File: Casks/diskmaker-x.rb [hunk 1]
```diff
@@ -1,6 +1,11 @@
 cask 'diskmaker-x' do
-  version '8.0.3'
-  sha256 '79b490dc829775450aafadeddd0afc58bdcef9c60fc82d9db1427c51b57e88a7'
+  if MacOS.version == :catalina
```
Comment (reitermarkus) [inline Casks/diskmaker-x.rb]: The other way round, use `MacOS.version <= :mojave` first.

Comment (reitermarkus) [reply Casks/diskmaker-x.rb]: Also the `app` needs to be in the `if` as well, since the app name differs.

**Comment by patbaumgartner** [comment]
Thanks for your feedback. Fixed it and hope it.

**Comment by reitermarkus** [comment]
Run `brew cask style --fix` again.

**Comment by reitermarkus** [comment]
The order of the `if` statement is wrong.

**Pull request merged**